pkgs/servers/firebird/default.nix

```diff
@@ -1,91 +1,84 @@
-   b) classic
-    - is built by default
-    - one process for each connection
-    - on linux direct io operations (?)
-   c) embedded.
-
-   manual says that you usually don't notice the difference between a and b.
-
-   I'm only interested in the embedder shared libary for now.
-   So everything isn't tested yet
-
-*/
-
-stdenv.mkDerivation rec {
-  version = "2.5.7.27050-0";
+{ lib, stdenv, fetchFromGitHub, libedit, autoreconfHook, zlib, unzip, libtommath, libtomcrypt, icu, superServer ? false }:
+
+let base = {
   pname = "firebird";
 
-  # enableParallelBuilding = false; build fails
+  meta = with lib; {
+    description = "SQL relational database management system";
+    homepage = "https://firebirdsql.org/";
+    changelog = "https://github.com/FirebirdSQL/firebird/blob/master/CHANGELOG.md";
+    license = [ "IDPL" "Interbase-1.0" ];
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ marcweber ];
+  };
+
+  nativeBuildInputs = [ autoreconfHook ];
+
+  buildInputs = [ libedit icu ];
 
-  # http://tracker.firebirdsql.org/browse/CORE-3246
-  preConfigure = ''
-    makeFlags="$makeFlags CPU=$NIX_BUILD_CORES"
+  LD_LIBRARY_PATH = lib.makeLibraryPath [ icu ];
+
+  configureFlags = [
+    "--with-system-editline"
+  ] ++ (lib.optional superServer "--enable-superserver");
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out
+    cp -r gen/Release/firebird/* $out
+    runHook postInstall
   '';
 
-  configureFlags =
-    [ "--with-serivec-port=${builtins.toString port}"
-      "--with-service-name=${serviceName}"
-      "--with-system-editline"
-      "--with-fblog=/var/log/firebird"
-      "--with-fbconf=/etc/firebird"
-      "--with-fbsecure-db=/var/db/firebird/system"
-    ]
-    ++ (lib.optional  (icu != null) "--with-system-icu")
-    ++ (lib.optional superServer "--enable-superserver");
-
-  src = fetchurl {
-    url = "mirror://sourceforge/firebird/Firebird-${version}.tar.bz2";
-    sha256 = "06hp6bq5irqvm3h03s79qjgcc3jsjpq150y9aq7anklx9v4nhfqa";
-  };
+}; in {
 
-  hardeningDisable = [ "format" ];
+  firebird_2_5 = stdenv.mkDerivation (base // rec {
+    version = "2.5.9";
 
-  # configurePhase = ''
-  #   sed -i 's@cp /usr/share/automake-.*@@' autogen.sh
-  #   sh autogen.sh $configureFlags --prefix=$out
-  # '';
-  buildInputs = [libedit icu automake autoconf libtool];
+    src = fetchFromGitHub {
+      owner = "FirebirdSQL";
+      repo = "firebird";
+      rev = "R${builtins.replaceStrings [ "." ] [ "_" ] version}";
+      sha256 = "sha256-YyvlMeBux80OpVhsCv+6IVxKXFRsgdr+1siupMR13JM=";
+    };
 
-  # TODO: Probably this hase to be tidied up..
-  # make install requires beeing. disabling the root checks
-  # dosen't work. Copying the files manually which can be found
-  # in ubuntu -dev -classic, -example packages:
-  # maybe some of those files can be removed again
-  installPhase = "cp -r gen/firebird $out";
+    configureFlags = base.configureFlags ++ [ "--with-system-icu" ];
```

Comment:
Would it make sense to instead
```nix
inherit (callPackages ../servers/firebird { icu = null; }) firebird_2_5;
```

Reply:
Will test.

Reply:
ICU null for Firebird 2.5 errored as:
```
IndicClassTables.cpp:229:117: error: narrowing conversion of '2147483651' from 'unsigned int' to 'icu_3_0::IndicClassTable::ScriptFlags' {aka 'int'} [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wnarrowing^G-Wnarrowing8;;^G]
  229 | static const IndicClassTable teluClassTable = {0x0C00, 0x0C6F, 3, TELU_SCRIPT_FLAGS, teluCharClasses, teluSplitTable};
```
Considering @MarcWeber comment, I understand Firebird 2.5 should be built with embedded ICU. But I could only build it using system-icu. I could be making something wrong. But I don't have a solution for  embedded ICU for now.

Reply:
One thing that possibly could work the same would be to provide ICU 3.0 if available.

Reply:
Then don't remove `(lib.optional  (icu != null) "--with-system-icu")`.

Reply:
There is no change to be made. I think the PR is ready.

Reply:
Maybe I could remove the superServer. To avoid load on hydra. Whoever needs it can override the package.

Reply:
To me PR is ready to be merged. Do you have any further considerations?

Reply:
I think I've misinterpreted your intention. Sorry for confusion. I thought the idea was to build the version 2.5.9 with the vendored icu (not from nixpkgs). But apparently it's not, which is fine by me.

Reply:
To clarify my intention: I'd have built Firebird 2.5.9 with vendored icu if my attempt was successful. As I failed to make it work. I'm leaving this possibility open. This should not stop still PR from being merged because that would mean making FB 2.5 (system ICU), FB 3, FB 4 (x86_64, aarch64) unavailable until the solution for embedded ICU problem is known.
---
Btw, thanks for reviewing the PR. Your review was helpful, I have learned some interesting bits. :-)